### ISSUE TITLE
Soft delete on a future date

### ISSUE THREAD

**Issue by MatheusRich** [ActsAsParanoid/acts_as_paranoid#138]
Can I set a future date to soft delete an object?

```ruby
object.deleted_at = DateTime.tomorrow
```

So the gem would check if `deleted_at != nil` and `DateTime.today < deleted_at`

**Comment by mvz** [comment]
No, whether an object is considered deleted only depends on whether `deleted_at` is `nil` or not.

**Comment by MatheusRich** [comment]
@marycodes2 do you guys have any interest in this feature? I could make a PR addressing this, if you give me some guidance.

**Comment by mvz** [comment]
Please be aware that you will want to have any callbacks related to the deletion run at that future date, and you would want to store somehow that the deletion has taken place.

Given those two constraints, I highly recommend using a mechanism external to ActsAsParanoid, that would use a different field or other storage for scheduling deletion. As a bonus, this would allow that mechanism to also work for hard-deleting objects.

In summary, I would not be in favor of adding this feature to ActsAsParanoid.

**Comment by mvz** [comment]
Check out https://github.com/jmettraux/rufus-scheduler for an existing scheduling solution that should be able to handle this.

**Comment by mvz** [comment]
TL;DR: you should schedule deletion outside of ActsAsParanoid.
